src/app/edit_config.rs

```diff
@@ -61,33 +57,39 @@ impl EditConfigState {
         self.is_editing = !self.is_editing;
     }
 
-    pub fn highlight_above_entry(&mut self) {
+    /// Move the highlight to the previous entry
+    pub fn highlight_prev(&mut self) {
         if self.highlighted_entry > 0 {
             self.highlighted_entry -= 1;
         }
     }
 
-    pub fn highlight_below_entry(&mut self) {
+    /// Move the highlight to the next entry
+    pub fn highlight_next(&mut self) {
         if self.highlighted_entry + 1 < self.config_buffer.len() {
             self.highlighted_entry += 1;
         }
     }
 
-    pub fn remove_char_from_editing_val(&mut self) {
+    /// Remove the last char from the current editing value if not empty
+    pub fn backspace_edit(&mut self) {
         if !self.editing_val.is_empty() {
             self.editing_val.pop();
         }
     }
 
-    pub fn add_char_to_editing_val(&mut self, ch: char) {
+    /// Appends a new char to the current editing value
+    pub fn append_edit(&mut self, ch: char) {
         self.editing_val.push(ch);
     }
 
-    pub fn clear_editing_val(&mut self) {
+    /// Clear the current editing value
+    pub fn clear_edit(&mut self) {
         self.editing_val.clear();
     }
 
-    pub fn push_editing_val_to_buffer(&mut self) {
+    /// Save the current edit value to the config buffer
+    pub fn save_edit(&mut self) {
```

Comment:
When building `edit_config`, I made a 2-step commit for configs, i.e.,

1. The user types in the new config value, which is only added to the buffer if the user hits ENTER (this is when we call this func)
2. After the user is pleased with the configs in the buffer, he/she hits ENTER to commit the buffer to the config file (and to the `Config` type at runtime).

So (sorry for keep on hitting on this tone) an unadvised reader won't be sure about what `config.save_edit()` means. I think something along the lines of `save_edit_to_buf` does the trick (if there is a word that gives the meaning of a "semi-commit", better). I know that the func doc further explains what it does and that in most setups, just a hover on the func will clear a doubt, but we shouldn't assume those things (also, this doesn't beat just reading the line of code and being "more sure" about what it does).

Reply:
Okay i'll think about a better name

Reply:
I was thinking here, what about `stage_edit` or `prepare_edit`?